manifests/init.pp

```diff
@@ -361,6 +364,36 @@
     validate_re($ssh_config_forward_x11_trusted_real, '^(yes|no)$', "ssh::ssh_config_forward_x11_trusted may be either 'yes' or 'no' and is set to <${ssh_config_forward_x11_trusted_real}>.")
   }
 
+  case type3x($sshd_x11_display_offset) {
+    'string': {
+      validate_re($sshd_x11_display_offset, '^\d+$', "ssh::sshd_x11_display_offset must be a valid number and is set to <${sshd_x11_display_offset}>.")
+      $sshd_x11_display_offset_array = [ str2num($sshd_x11_display_offset) ]
+    }
+    'integer': {
+      $sshd_x11_display_offset_array = [ $sshd_x11_display_offset ]
+    }
+    default: {
+      fail('ssh:sshd_x11_display_offset must be a string or an integer. ')
+    }
+  }
+
+  case type3x($sshd_key_regeneration_interval) {
+    'string': {
+      validate_re($sshd_key_regeneration_interval, '^\d+$', "ssh::sshd_key_regeneration_interval must be a valid number and is set to <${sshd_key_regeneration_interval}>.")
```

Comment:
Suggest this pattern which if the variable is an integer, it will be treated as a string for validate_re()

```puppet
if $sshd_x11_display_offset != undef {
  validate_re("${sshd_x11_display_offset}", '^\d+$', "ssh::sshd_x11_display_offset must be a valid number and is set to <${sshd_x11_display_offset}>.") # lint:ignore:only_variable_string
}
```